src/go/configgenerator/listener_generator.go

```diff
@@ -50,8 +50,87 @@ const (
 	statPrefix = "ingress_http"
 )
 
+// MakeListeners provides dynamic listeners for Envoy
+func MakeListeners(serviceInfo *sc.ServiceInfo) ([]*v2pb.Listener, error) {
+	var listeners []*v2pb.Listener
+
+	port := serviceInfo.Options.ListenerPort
+	isSsl := false
+	if serviceInfo.Options.SslPort != 0 {
+		port = serviceInfo.Options.SslPort
+		isSsl = true
+	}
+
+	listener, err := makeListener(serviceInfo, port, isSsl)
+	if err != nil {
+		return nil, err
+	}
+	listeners = append(listeners, listener)
+
+	redirect_listener, err := makeRedirectListener(serviceInfo)
+	if err != nil {
+		return nil, err
+	}
+	if serviceInfo.Options.SslPort != 0 {
+		listeners = append(listeners, redirect_listener)
+	}
+
+	return listeners, nil
+}
+
+func makeRedirectListener(serviceInfo *sc.ServiceInfo) (*v2pb.Listener, error) {
```

Comment:
Maybe move `makeRedirectListener` into MakeListener like other listeners because they are parallel,instead of moving all other listeners into `makeListener`?

Reply:
No, there is only one Listener right now.

This RedirectListener is in parallel to the existing listener.